flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableSource.java

```diff
@@ -49,27 +52,42 @@
 /**
  * A TableSource implementation to read data from Hive tables.
  */
-public class HiveTableSource extends InputFormatTableSource<Row> {
+public class HiveTableSource extends InputFormatTableSource<Row> implements PartitionableTableSource {
 
 	private static Logger logger = LoggerFactory.getLogger(HiveTableSource.class);
 
 	private final JobConf jobConf;
 	private final ObjectPath tablePath;
 	private final CatalogTable catalogTable;
-	private List<HiveTablePartition> allPartitions;
+	private List<HiveTablePartition> allHivePartitions;
 	private String hiveVersion;
+	//partitionList represent all partitions in map list format used in partition-pruning situation.
+	private List<Map<String, String>> partitionList = new ArrayList<>();
+	private Map<Map<String, String>, HiveTablePartition> partitionSpec2HiveTablePartition = new HashMap<>();
 
 	public HiveTableSource(JobConf jobConf, ObjectPath tablePath, CatalogTable catalogTable) {
 		this.jobConf = jobConf;
 		this.tablePath = tablePath;
 		this.catalogTable = catalogTable;
 		this.hiveVersion = jobConf.get(HiveCatalogValidator.CATALOG_HIVE_VERSION, HiveShimLoader.getHiveVersion());
+		initAllPartitions();
+	}
+
+	private HiveTableSource(JobConf jobConf, ObjectPath tablePath, CatalogTable catalogTable,
+							List<HiveTablePartition> allHivePartitions,
+							String hiveVersion,
+							List<Map<String, String>> partitionList) {
+		this.jobConf = jobConf;
+		this.tablePath = tablePath;
+		this.catalogTable = catalogTable;
+		this.allHivePartitions = allHivePartitions;
+		this.hiveVersion = hiveVersion;
+		this.partitionList = partitionList;
 	}
 
 	@Override
 	public InputFormat getInputFormat() {
-		initAllPartitions();
-		return new HiveTableInputFormat(jobConf, catalogTable, allPartitions);
+		return new HiveTableInputFormat(jobConf, catalogTable, allHivePartitions);
```

Comment:
I'm not sure of the call orders, but wondering if it makes sense to create a HiveTableInputFormat still with all the partitions. Maybe some partitions are pruned out.

Reply:
I'd prefer not. We can keep the partition pruning logic clean in `HiveTableSource` and `HiveInputFormat` logic clean.
I add a flag named `initAllPartitions` to avoid potential bugs invoked by call orders.